### DISCUSSION TITLE
how to migrate from serve_addr_handle in hyper 0.14.4?

### DISCUSSION THREAD

**Discussion by kushwahashiv** [hyperium/hyper#2455] [Q&A]
I am trying to migrate below code to new hyper `0.14.4` version  

```rust
Http::new().serve_addr_handle(&address, &handle, move || {
            // Prepare application
             ...
         Ok(app)
})
```

current dependencies
```rust
futures-core = { version = "0.3.12" }
futures-util = { version = "0.3.12" },
futures-channel = "0.3.12"
http = "0.2.3"
tokio = { version = "1.2.0" }
hyper = { version = "0.14.4", features = ["full"]}
```
but facing challenges what is the best way to migrate this.

If anyone has done that already what are the best practices. If possible to keep it close to the above implementation would be great.

Also if there is any link would be helpful. 

Thanks in advance :)

/Shiv

**Comment by seanmonstar** [comment]
If you want to easily listen to an address and have hyper handle that, then you probably could switch to using [`hyper::Server`](https://docs.rs/hyper/0.14.*/hyper/server/struct.Server.html):

```rust
hyper::Server::bind(addr)
    // .options(args)
    .serve(make_service_fn(move |_socket_ref| async move {
        // create a Service 
        // ...
        Ok(svc)
    })
    .await;
```
